### PR TITLE
ISPN-8641 Wildfly 11 Upgrade

### PR DESCRIPTION
**Pull request by ryanemerson**
https://issues.jboss.org/browse/ISPN-8641

~This builds upon the work in https://github.com/infinispan/infinispan/pull/5652 which must be integrated first.~

~This is very much a preview at the moment to see what I have broke.~ Note, so far I have only utilised Elytron where necessary (i.e. to replace jboss-sasl), but long term we should be able to refactor various other parts of our code to take advantage of Elytron.

### CODE REVIEW

File: server/integration/endpoint/src/test/java/org/infinispan/server/endpoint/EndpointSubsystemTestCase.java [hunk 1]
```diff
@@ -69,10 +69,10 @@ public EndpointSubsystemTestCase(String xmlFile, int operations, String xsdPath,
    @Parameters
    public static Collection<Object[]> data() {
       Object[][] data = new Object[][] {
-            { "endpoint-7.2.xml", 16, "schema/jboss-infinispan-endpoint_7_2.xsd", null },
-            { "endpoint-8.0.xml", 16, "schema/jboss-infinispan-endpoint_8_0.xsd", null },
-            { "endpoint-9.0.xml", 38, "schema/jboss-infinispan-endpoint_9_0.xsd", null },
-            { "endpoint-9.2.xml", 38, "schema/jboss-infinispan-endpoint_9_2.xsd", new String[] { "/subsystem-templates/infinispan-endpoint.xml"} },
+            { "endpoint-7.2.xml", 15, "schema/jboss-infinispan-endpoint_7_2.xsd", null },
+            { "endpoint-8.0.xml", 15, "schema/jboss-infinispan-endpoint_8_0.xsd", null },
+            { "endpoint-9.0.xml", 37, "schema/jboss-infinispan-endpoint_9_0.xsd", null },
```
Comment (gustavonalle) [inline server/integration/endpoint/src/test/java/org/infinispan/server/endpoint/EndpointSubsystemTestCase.java]: In my understanding, the old config files and old xsds didn't change, so this count should remain the same?

Comment (ryanemerson) [reply server/integration/endpoint/src/test/java/org/infinispan/server/endpoint/EndpointSubsystemTestCase.java]: The count is the number of executed operations, so removing the "com.sun.security.sasl.digest.utf8" property means that the count must be decremented by one.

Comment (gustavonalle) [reply server/integration/endpoint/src/test/java/org/infinispan/server/endpoint/EndpointSubsystemTestCase.java]: Got it, thanks

**Comment by gustavonalle** [comment]
Build failed on CI, after generating 10Gb of console logs. 
I run the build locally, and it went well, I only found out commons-http was in TRACE mode so all rest tests were a bit more verbose, but all in all the console log was ~100 Mb.

❓ ❓

**Comment by gustavonalle** [comment]
btw, couldn't figure out where jenkins keep console logs

**Comment by ryanemerson** [comment]
https://github.com/infinispan/infinispan/pull/5656/commits/7b61c8a500db513f443623ecb984d103d85329b6 can be removed once https://github.com/infinispan/infinispan/pull/5720 is merged.

**Comment by gustavonalle** [comment]
I inspected one of the large log files, and the test suite proceeds normally until it hits the test 
```RESTClientWithSniEncryptionIT```. 

From this point on, when the ```org.infinispan.server.test.client.rest.RESTHelper``` is first used, it starts to use the "wire" log of [apache commons HTTP ](https://hc.apache.org/httpcomponents-client-ga/logging.html)

It then proceeds until ```AbstractRESTClientIT#testStoreBigObject``` where each byte is printed
```
[org.apache.http.wire] : http-outgoing-6 >> "[0x0][0x1][0x2][0x3][0x4] "
```

I keeps running that request several times not sure why as this test should run "only" 3 times (there is 3 
classes inheriting from ```AbstractRESTClientIT```

**Comment by slaskawi** [comment]
@gustavonalle @ryanemerson Does the `RESTClientWithSniEncryptionIT` test pass? Or this is a mixture of failures and diagnostic runs?

**Comment by gustavonalle** [comment]
@slaskawi it passes

**Comment by slaskawi** [comment]
@gustavonalle oooh... hmmm. So I guess there is no other way than attaching the debugging and checking what's turning this logging on....

**Comment by gustavonalle** [comment]
Taking only the first GB of the 17 GB log file, it runs ```RESTClientIT.testStoreBigObject```  23 times, logging each byte to the console. I wonder who is retrying that test so many times

**Comment by gustavonalle** [comment]
@ryanemerson I was able to silence that logger by adding 
```xml
 <dependency>
   <groupId>org.apache.logging.log4j</groupId>
   <artifactId>log4j-jcl</artifactId>
</dependency>
```

to ```server/testsuite/pom.xml```, would you like to give a try?

**Comment by ryanemerson** [comment]
@gustavonalle I think the problem is `jcl-over-slf4j` was removed. I honestly can't remember why I did this, I'm investigating atm.

**Comment by ryanemerson** [comment]
The latest changes should fix the logging issue

**Comment by gustavonalle** [comment]
Integrated, thanks @ryanemerson !

**Pull request merged**